R/model-tools.R

```diff
@@ -302,44 +307,43 @@ enw_sample <- function(data, model = epinowcast::enw_model(),
 #' @importFrom cmdstanr cmdstan_model
 #' @examplesIf interactive()
 #' mod <- enw_model()
-enw_model <- function(model, include, compile = TRUE,
-                      threads = FALSE, profile = FALSE,
-                      stanc_options = list(), verbose = TRUE, ...) {
-  if (missing(model)) {
-    model <- "stan/epinowcast.stan"
-    model <- system.file(model, package = "epinowcast")
-  }
-  if (missing(include)) {
-    include <- system.file("stan", package = "epinowcast")
+enw_model <- function(
+    model = system.file("stan", "epinowcast.stan", package = "epinowcast"),
+    include = dirname(model),
+    compile = TRUE, threads = FALSE, profile = FALSE,
+    stanc_options = list(), verbose = TRUE,
+    ...
+) {
+
+  if (verbose) {
+    message(sprintf("Using model %s.", model))
+    message(sprintf("include is %s.", paste(include, collapse = ", ")))
   }
 
   if (!profile) {
     stan_no_profile <- write_stan_files_no_profile(model, include)
     model <- stan_no_profile$model
     include <- stan_no_profile$include_paths
+    if (verbose) {
```

Comment:
I don't think we want the messaging this way around. I think it should message if profiling is included (i.e not the default) as that is something users only really want when doing development work.

Reply:
mmm, then i think just remove. i only had it in there to run down some issues with `cmdstanr` and include paths (the solution to which was "install latest")